public/app/features/explore/RichHistory/RichHistoryStarredTab.tsx

```diff
@@ -87,9 +86,10 @@ export function RichHistoryStarredTab(props: Props) {
   const listOfDatasourceFilters = datasourceFilters?.map(d => d.value);
 
   const starredQueries = queries.filter(q => q.starred === true);
-  const starredQueriesFilteredByDatasource = datasourceFilters
-    ? starredQueries?.filter(q => listOfDatasourceFilters?.includes(q.datasourceName))
-    : starredQueries;
```

Comment:
Updated as new Select for empty selected list uses [] instead of null (which was used by LegacySelect)